src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeService.java

```diff
@@ -65,7 +76,18 @@ private UUID createEnvelope(String containerName, String blobName, Instant blobC
     }
 
     @Transactional
-    public void markEnvelopeAsDeleted(UUID envelopeId) {
-        envelopeRepository.markAsDeleted(envelopeId);
+    public void markEnvelopeAsDeleted(Envelope envelope) {
+        envelopeRepository.markAsDeleted(envelope.id);
+        eventRecordRepository.insert(new NewEventRecord(envelope.container, envelope.fileName, Event.DELETED));
+    }
+
+    @Transactional
+    public void eventForProcessStart(String containerName, String blobName) {
+        eventRecordRepository.insert(new NewEventRecord(containerName, blobName, Event.FILE_PROCESSING_STARTED));
+    }
+
+    @Transactional
+    public void eventForDeletionFromRejected(String containerName, String blobName) {
```

Comment:
weird for a method that doesn't return anything to not have a verb. Maybe `saveEventXXX(...)`?

Reply:
at that moment no name got my attention - left here. i'll change to saveEventSmth. most likely not xxx though 😆